### PR TITLE
Fix lazy init of `BatchNormalization` link

### PR DESCRIPTION
**Pull request by hvy**
Fixes https://github.com/chainer/chainer/issues/6368.

### CODE REVIEW

File: tests/chainer_tests/links_tests/normalization_tests/test_batch_normalization.py [hunk 1]
```diff
@@ -504,4 +505,39 @@ def test_fail_inference(self):
             links.BatchNormalization()
 
 
+@attr.multi_gpu(2)
+class TestLazyInitializationWithNonZeroCurrentCudaDevice(unittest.TestCase):
+
+    def test_lazy_initialization_with_non_zero_current_cuda_device(self):
+        x = numpy.random.randn(5, 4, 3, 2).astype('f')
```
Comment (niboshi) [inline tests/chainer_tests/links_tests/normalization_tests/test_batch_normalization.py]: Move this line to just above `x = device.send(x)`. It's not used before that.

Comment (niboshi) [inline tests/chainer_tests/links_tests/normalization_tests/test_batch_normalization.py]: Dtype char code should be avoided (discouraged in numpy documentation).

Comment (hvy) [reply tests/chainer_tests/links_tests/normalization_tests/test_batch_normalization.py]: Fixed.

Comment (hvy) [reply tests/chainer_tests/links_tests/normalization_tests/test_batch_normalization.py]: Thanks, fixed.

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
LGTM

**Pull request merged**